main.py

```diff
@@ -129,14 +129,13 @@ def run_scenario(current_scenario):
         current_scenario.split_data()
     current_scenario.plot_data_distribution()
     current_scenario.compute_batch_sizes()
-    current_scenario = fl_training.preprocess_scenarios_data(current_scenario)
+    current_scenario = utils.preprocess_scenarios_data(current_scenario)
 
     # Train and eval on all partners according to scenario
     is_save_fig = True
     start = timer()
-    current_scenario.federated_test_score = fl_training.compute_test_score_with_scenario(
-        current_scenario, is_save_fig
-    )
+    mpl = multi_partner_learning.init_multipartnerlearning_from_scenario(current_scenario)
+    current_scenario.federated_test_result = mpl.compute_test_score()
```

Comment:
mpl could be an attribute of the scenario instance. This way it would be easier to get back results from the mpl instance to the scenario instance.